File: ruby/reverse_string.rb
#26 question on interview cake

def reverse_string(str)
  left_pointer = 0
  right_pointer = str.length - 1

  while left_pointer < right_pointer
    temp = str[left_pointer]
    str[left_pointer] = str[right_pointer]
    str[right_pointer] = temp

    left_pointer += 1
    right_pointer -= 1
  end
  str
end

File: javascript/reverse_string.js
// #26 on interview cake

function reverse_string(str){
  var wordArr = str.split('');

  var temp;
  var startIndex = 0;
  var endIndex = wordArr.length;

  while (startIndex < endIndex){
    temp = wordArr[startIndex]
    wordArr[startIndex] = wordArr[endIndex]
    wordArr[endIndex] = temp;

    startIndex++
    endIndex--
  }
  return wordArr.join('')
}
